lib/default.nix

```diff
@@ -204,6 +204,18 @@ let
         ) (lib.attrsToList hosts)
       );
 
+      # Check if the given module is wrapped in a function expecting "publisher" as a named argument,
+      # and - if so - call that function with the current flake. This enables us to reference
+      # the publishing flake when re-using exported modules in consuming, downstream flakes.
+      withPublisher =
+        module:
+        let
+          module' = if builtins.isString module || builtins.isPath module then import module else module;
+          wantsPublisher =
+            builtins.isFunction module' && builtins.hasAttr "publisher" (builtins.functionArgs module');
```

Comment:
What do you think of applying the same logic for all the specialArgs? `flake`, `inputs`, `perSystem`. As a module author, I don't think I ever wanted to get access to inputs that are not part of the current flake.

Reply:
Not entirely sure, I understand this proposal correctly:

* If you mean changing the wrapping functions arguments to 
`flake`, `inputs`, `perSystem` instead of `publisher`. Could do that but the current design was meant to a) keep the api surface small and clearly named and b) still allow people to use the module args `flake`, `inputs`, `perSystem` to refer to the current/consuming flake.

* If you mean not as an addition, but moving those args to the wrapping function all together instead of module arguments: That would make it difficult to access i.e. packages from the current/consum flake, no?